avm/res/network/firewall-policy/tests/e2e/max/main.test.bicep

```diff
@@ -0,0 +1,91 @@
+targetScope = 'subscription'
+
+metadata name = 'Using large parameter set'
+metadata description = 'This instance deploys the module with most of its features enabled.'
+
+// ========== //
+// Parameters //
+// ========== //
+
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(90)
+param resourceGroupName string = 'dep-${namePrefix}-network.firewallpolicies-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'nfpmax'
+
+@description('Optional. A token to inject into the name of each resource.')
+param namePrefix string = '#_namePrefix_#'
+
+// ============ //
+// Dependencies //
+// ============ //
+
+// General resources
+// =================
+resource resourceGroup 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+  name: resourceGroupName
+  location: location
+}
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
```

Comment:
Should contain an MSI test I pressume

Reply:
I agree, however the Managed Identity is part of the TLS Inspection, which is a Premium feature. 
We could switch the the `max` test case to Premium tier. 
Actually it would be even better to have a separate test case with the Premium features, but the TLS Inspection requires a certificate with particular properties, which is not that easy to generate 'on-the-fly' for test purposes.
Also for now, as a part of the module migration I would propose to just test the MSI using the `max` test case and Premium tier. And the full test of all Premium features like TLS Inspection or IDPS (intrusion detection and prevention system) could be implemented later. What do you think?

Reply:
Added: https://github.com/Azure/bicep-registry-modules/pull/767/commits/afdce4a052437c35896c01953d5742aff4b12c27

Reply:
Good Effort @krbar - This is also similar for Service Bus we used the Max test to test the premium features. I would state I would only expect to see a separate test if there is entirely different parameter set functionality between the SKU's e.g